### ISSUE TITLE
SA1028 (CodeMustNotContainTrailingWhitespace) tests are prone to regression errors

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#797]
The source file containing the unit tests for SA1028 relies on lines that themselves contain trailing whitespace. This file should avoid any such lines to ensure that neither this code fix nor a user's customized development environment alters the file in such a way that the meaning of the tests is lost.

**Comment by AArnott** [comment]
The trailing whitespace used in the tests are within a @"" string, so no tool I've seen will "fix" and remove that whitespace. And if it did, it would immediately cause the test to fail (unless the product simultaneously regressed in a precisely matching way). 
If we want to avoid trailing whitespace as seen by tools that don't recognize C#, I imagine we could do that, but that would make the test code harder to read and maintain IMO.

Unless I'm missing something, I recommend a Won't Fix on this (or By Design, if you prefer).

**Comment by sharwell** [comment]
For this particular file, I would prefer verbatim string literals be avoided in favor of regular string literals. This would eliminate the chance that the analyzer/code fix would ever make an automatic change to its own test suite. It also improves clarity for users reading through the code, since it's not easy to see what the test is actually doing in some viewers (e.g. even looking on this site is a challenge).

**Comment by EdwinEngelen** [comment]
grabbed

**Comment by sharwell** [comment]
@EdwinEngelen Welcome to DotNetAnalyzers! I've assigned the issue to you. Let us know if you have any questions or run into problems. :+1:

**Comment by EdwinEngelen** [comment]
Style 1:
            var testCodeBuilder = new StringBuilder();
            testCodeBuilder.AppendLine("class ClassName");
            testCodeBuilder.AppendLine("{");
            testCodeBuilder.AppendLine("    void MethodName()");
            testCodeBuilder.AppendLine("    {");
            testCodeBuilder.AppendLine("        System.Console.WriteLine(); ");
            testCodeBuilder.AppendLine("    }");
            testCodeBuilder.AppendLine("}");
            string testCode = testCodeBuilder.ToString();

Style 2:
            var testCode =
                "class ClassName" + Environment.NewLine +
                "{" + Environment.NewLine +
                "    void MethodName()" + Environment.NewLine +
                "    {" + Environment.NewLine +
                "        System.Console.WriteLine(); " + Environment.NewLine +
                "    }" + Environment.NewLine +
                "}" + Environment.NewLine;

In my opinion the test code is harder to read with style 2 due to the 'Environment.NewLine'.

**Comment by sharwell** [comment]
What about this?

```csharp
var testCode = new StringBuilder()
    .AppendLine("class ClassName")
    .AppendLine("{")
    .ToString();
```

**Comment by EdwinEngelen** [comment]
You're great. Didn't know about the return value. (y)

**Comment by sharwell** [comment]
Fixed in #990.
